pkg/parser/ast/dml.go

```diff
@@ -707,6 +711,11 @@ type SelectField struct {
 	Auxiliary             bool
 	AuxiliaryColInAgg     bool
 	AuxiliaryColInOrderBy bool
+
+	// IsUnfoldFromWildCard indicates whether this field is unfolded from a wildcard, which can be used in checking privilege.
+	// Although we always check SELECT privilege in column-level, a table-level access deny error will be return if the
```

Comment:
Is this the logic in the design? Or is it the same as the MySQL logic? Can we add an explanation?

Reply:
Yes. Assume that table `t`  has columns `(a,b,c)`
Both `SELECT * FROM t` and `SELECT a,b,c FROM t` require `SELECT` privilege of `a`,`b` and `c`.
But if lacking privilege, the former reports error code `1142`, the latter reports `1143`.